File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.IOException;
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
    static ArrayList<Integer>[] graph;
    static boolean[] visited;
    static int N;
    static int M;
    static int cnt = 0;
    static void dfs(int v) {
        for (int i : graph[v]) {
            if(!visited[i]) {
                visited[i] = true;
                cnt += 1;
                dfs(i);
            }
        }
    }

    public static void main(String[] args) throws IOException {
        Scanner s = new Scanner(System.in);
        N = s.nextInt();
        M = s.nextInt();
        visited = new boolean[N+1];
        graph = new ArrayList[N+1];
        for(int i = 0; i < N + 1; i++) {
            graph[i] = new ArrayList<>();
        }
        for (int k = 0; k < M; k++) {
            int i = s.nextInt();
            int j = s.nextInt();
            graph[i].add(j);
            graph[j].add(i);
        }
        visited[1] = true;
        dfs(1);
        System.out.println(cnt);
        s.close();
    }
}

File: 백준/Silver/2606. 바이러스/바이러스.py
N = int(input())
M = int(input())
graph = [[] for _ in range(N+1)]

visited = [False]*(N+1)
for _ in range(M):
    i, j = map(int,input().split())
    graph[i].append(j)
    graph[j].append(i)

def dfs(i):
    global count
    for j in graph[i]:
        if not visited[j]:
            visited[j] = True
            dfs(j)
            count += 1
visited[0] = True
visited[1] = True
count = 0
dfs(1)
print(count)
    
   